userspace/libsinsp/container.h

```diff
@@ -122,11 +123,49 @@ class sinsp_container_manager
 	void create_engines();
 	void cleanup();
 
+	void set_docker_socket_path(std::string socket_path);
 	void set_query_docker_image_info(bool query_image_info);
 	void set_cri_extra_queries(bool extra_queries);
 	void set_cri_socket_path(const std::string& path);
 	void set_cri_timeout(int64_t timeout_ms);
 	sinsp* get_inspector() { return m_inspector; }
+
+	/**
+	 * \brief set the status of an async container metadata lookup
+	 * @param container_id the container id we're looking up
+	 * @param ctype the container engine that is doing the lookup
+	 * @param state the state of the lookup
+	 *
+	 * Container engines that do not do any lookups in external services need not
+	 * bother with this. Otherwise, the engine needs to maintain the current
+	 * state of the lookup via this method and call should_lookup() before
+	 * starting a new lookup.
+	 */
+	void set_lookup_status(const std::string& container_id, sinsp_container_type ctype, sinsp_container_lookup_state state)
```

Comment:
Why does only docker use this, shouldn't cri use it as well? Or will that come only with cri async.